src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -799,7 +818,11 @@ private static SuggestedActionSetPriority GetSuggestedActionSetPriority(CodeActi
                             document, selection, isBlocking: true, addOperationScope, cancellationToken),
                         cancellationToken).WaitAndGetResult(cancellationToken);
 
-                    var filteredRefactorings = FilterOnUIThread(refactorings, workspace);
+                    // If we are computing refactorings outside the 'Refactoring' context, i.e. for example, from the lightbulb under a squiggle or selection,
+                    // then we want to filter out refactorings outside the selection span.
```

Comment:
i would explain 'why' (or link to an issue).

Reply:
Added a note in https://github.com/dotnet/roslyn/issues/44855